cmd/kubeadm/app/apis/kubeadm/v1alpha3/conversion.go

```diff
@@ -28,6 +29,10 @@ func Convert_v1alpha3_JoinConfiguration_To_kubeadm_JoinConfiguration(in *JoinCon
 		return err
 	}
 
+	if len(in.ClusterName) != 0 {
+		return errors.New("clusterName has been removed from JoinConfiguration and clusterName from ClusterConfiguration will be used instead. Please cleanup JoinConfiguration.ClusterName fields")
```

Comment:
i don't think this is needed, but will defer to @fabriziopandini for second opinion.
also PSA that the "global" yaml parser we are using will "soon" going to start throw errors on unknown and duplicate fields. hopefully in 1.13.

Reply:
I am ok with this